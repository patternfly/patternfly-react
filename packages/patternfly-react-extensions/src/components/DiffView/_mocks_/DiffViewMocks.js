import React from 'react';

export const diffMock = {
  oldText: 'hello friend',
  newText: 'hello there friend',
  viewType: 'split',
  emptyState: <div>empty</div>
};

export const patchMock = {
  viewType: 'unified',
  patch: '---',
  emptyState: <div>empty</div>
};

export const emptyState = {
  oldText: 'hello friend',
  newText: 'hello friend',
  viewType: 'split',
  emptyState: <div>empty</div>
};
