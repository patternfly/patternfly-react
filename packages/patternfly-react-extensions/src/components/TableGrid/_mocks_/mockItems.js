export const mockItems = [
  {
    title: 'Item 1',
    description: 'This is Item 1 description',
    hosts: 3,
    clusters: 1
  },
  {
    title: 'Item 2',
    description: 'This is Item 2 description',
    hosts: 10,
    clusters: 2
  },
  {
    title: 'Item 3',
    description: 'This is Item 3 description',
    hosts: 13,
    clusters: 11
  },
  {
    title: 'Item 4',
    description: 'This is a long description for item 4 that is longer than any other description.',
    hosts: 7,
    clusters: 3
  },
  {
    title: 'Item 5 with a very long title, showing that it should be truncated when it gets to be to long.',
    description: 'This is Item 5 description',
    hosts: 2,
    clusters: 9
  },
  {
    title: 'Item 6',
    description: 'This is Item 6 description',
    hosts: 1,
    clusters: 0
  }
];
