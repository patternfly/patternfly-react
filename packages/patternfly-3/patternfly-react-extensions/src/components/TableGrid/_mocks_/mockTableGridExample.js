import React from 'react';
import PropTypes from 'prop-types';
import TableGrid from '../TableGrid';
import { Grid, OverlayTrigger, Tooltip } from 'patternfly-react';
import { mockItems } from './mockItems';

const titleColSizes = {
  md: 2,
  sm: 4,
  xs: 6
};
const descrColSizes = {
  md: 6,
  sm: 4,
  xs: 6
};
const countColSizes = {
  md: 2,
  sm: 2,
  xsHidden: true
};

class MockTableGridExample extends React.Component {
  state = {
    sortField: 'title',
    isAscending: true,
    items: mockItems
  };

  onSortToggle = id => {
    const { items, sortField, isAscending } = this.state;
    let updateAscending = true;

    if (id === sortField) {
      updateAscending = !isAscending;
    }

    items.sort((a, b) => {
      let compVal = 0;
      if (id === 'title') {
        compVal = a.title.localeCompare(b.title);
      } else if (id === 'hosts') {
        compVal = a.hosts - b.hosts;
      } else if (id === 'clusters') {
        compVal = a.clusters - b.clusters;
      }

      if (!updateAscending) {
        compVal *= -1;
      }

      return compVal;
    });

    this.setState({ items, sortField: id, isAscending: updateAscending });
  };

  renderItemRow = (item, index) => (
    <TableGrid.Row key={index}>
      <Grid.Col {...titleColSizes}>{item.title}</Grid.Col>
      <Grid.Col {...descrColSizes}>{item.description}</Grid.Col>
      <Grid.Col {...countColSizes}>{item.hosts}</Grid.Col>
      <Grid.Col {...countColSizes}>{item.clusters}</Grid.Col>
    </TableGrid.Row>
  );

  render() {
    const { items, sortField, isAscending } = this.state;
    const { bordered } = this.props;
    return (
      <TableGrid id="table-grid" bordered={bordered}>
        <TableGrid.Head>
          <TableGrid.ColumnHeader
            id="title"
            sortable
            isSorted={sortField === 'title'}
            isAscending={isAscending}
            onSortToggle={() => this.onSortToggle('title')}
            {...titleColSizes}
          >
            Title
          </TableGrid.ColumnHeader>
          <TableGrid.ColumnHeader
            id="description"
            isSorted={sortField === 'description'}
            isAscending={isAscending}
            onSortToggle={() => this.onSortToggle('description')}
            {...descrColSizes}
          >
            Description
          </TableGrid.ColumnHeader>
          <TableGrid.ColumnHeader
            id="hosts"
            sortable
            isSorted={sortField === 'hosts'}
            isAscending={isAscending}
            onSortToggle={() => this.onSortToggle('hosts')}
            {...countColSizes}
          >
            <OverlayTrigger overlay={<Tooltip id="hosts-tip">Hosts</Tooltip>} placement="top">
              <span>Hosts</span>
            </OverlayTrigger>
          </TableGrid.ColumnHeader>
          <TableGrid.ColumnHeader
            id="clusters"
            sortable
            isSorted={sortField === 'clusters'}
            isAscending={isAscending}
            onSortToggle={() => this.onSortToggle('clusters')}
            {...countColSizes}
          >
            <OverlayTrigger overlay={<Tooltip id="clusters-tip">Clusters</Tooltip>} placement="top">
              <span>Clusters</span>
            </OverlayTrigger>
          </TableGrid.ColumnHeader>
        </TableGrid.Head>
        <TableGrid.Body>{items.map((item, index) => this.renderItemRow(item, index))}</TableGrid.Body>
      </TableGrid>
    );
  }
}

MockTableGridExample.propTypes = {
  bordered: PropTypes.bool
};

MockTableGridExample.defaultProps = {
  bordered: false
};

export { MockTableGridExample };

export const MockTableGridExampleSource = `
import React from 'react';
import PropTypes from 'prop-types';
import TableGrid from '../TableGrid';
import { Grid, OverlayTrigger, Tooltip } from 'patternfly-react';
import { mockItems } from './mockItems';

const titleColSizes = {
  md: 2,
  sm: 4,
  xs: 6
};
const descrColSizes = {
  md: 6,
  sm: 4,
  xs: 6
};
const countColSizes = {
  md: 2,
  sm: 2,
  xsHidden: true
};

class MockTableGridExample extends React.Component {
  state = {
    sortField: 'title',
    isAscending: true,
    items: mockItems
  };

  onSortToggle = id => {
    const { items, sortField, isAscending } = this.state;
    let updateAscending = true;

    if (id === sortField) {
      updateAscending = !isAscending;
    }

    items.sort((a, b) => {
      let compVal = 0;
      if (id === 'title') {
        compVal = a.title.localeCompare(b.title);
      } else if (id === 'hosts') {
        compVal = a.hosts - b.hosts;
      } else if (id === 'clusters') {
        compVal = a.clusters - b.clusters;
      }

      if (!updateAscending) {
        compVal *= -1;
      }

      return compVal;
    });

    this.setState({ items, sortField: id, isAscending: updateAscending });
  };

  renderItemRow = (item, index) => (
    <TableGrid.Row key={index}>
      <Grid.Col {...titleColSizes}>{item.title}</Grid.Col>
      <Grid.Col {...descrColSizes}>{item.description}</Grid.Col>
      <Grid.Col {...countColSizes}>{item.hosts}</Grid.Col>
      <Grid.Col {...countColSizes}>{item.clusters}</Grid.Col>
    </TableGrid.Row>
  );

  render() {
    const { items, sortField, isAscending } = this.state;
    const { bordered } = this.props;
    return (
      <TableGrid id="table-grid" bordered={bordered}>
        <TableGrid.Head>
          <TableGrid.ColumnHeader
            id="title"
            sortable
            isSorted={sortField === 'title'}
            isAscending={isAscending}
            onSortToggle={() => this.onSortToggle('title')}
            {...titleColSizes}
          >
            Title
          </TableGrid.ColumnHeader>
          <TableGrid.ColumnHeader
            id="description"
            isSorted={sortField === 'description'}
            isAscending={isAscending}
            onSortToggle={() => this.onSortToggle('description')}
            {...descrColSizes}
          >
            Description
          </TableGrid.ColumnHeader>
          <TableGrid.ColumnHeader
            id="hosts"
            sortable
            isSorted={sortField === 'hosts'}
            isAscending={isAscending}
            onSortToggle={() => this.onSortToggle('hosts')}
            {...countColSizes}
          >
            <OverlayTrigger overlay={<Tooltip id="hosts-tip">Hosts</Tooltip>} placement="top">
              <span>Hosts</span>
            </OverlayTrigger>
          </TableGrid.ColumnHeader>
          <TableGrid.ColumnHeader
            id="clusters"
            sortable
            isSorted={sortField === 'clusters'}
            isAscending={isAscending}
            onSortToggle={() => this.onSortToggle('clusters')}
            {...countColSizes}
          >
            <OverlayTrigger overlay={<Tooltip id="clusters-tip">Clusters</Tooltip>} placement="top">
              <span>Clusters</span>
            </OverlayTrigger>
          </TableGrid.ColumnHeader>
        </TableGrid.Head>
        <TableGrid.Body>{items.map((item, index) => this.renderItemRow(item, index))}</TableGrid.Body>
      </TableGrid>
    );
  }
}

MockTableGridExample.propTypes = {
  bordered: PropTypes.bool
};

MockTableGridExample.defaultProps = {
  bordered: false
};

export { MockTableGridExample };
`;
