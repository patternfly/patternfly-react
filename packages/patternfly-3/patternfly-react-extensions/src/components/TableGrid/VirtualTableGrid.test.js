import React from 'react';
import { mount } from 'enzyme';

import { VirtualTableGrid } from './index';
import { mockVirtualItems } from './_mocks_/mockVirtualItems';

const titleColSizes = {
  md: 2,
  sm: 4,
  xs: 6
};
const descrColSizes = {
  md: 6,
  sm: 4,
  xs: 6
};
const countColSizes = {
  md: 2,
  sm: 2,
  xsHidden: true
};

const ItemHeader = tableData => {
  const { selectType, selectedItems, sortField, isAscending, numItems } = tableData;
  return (
    <VirtualTableGrid.Head
      showCheckbox={selectType === 'checkbox'}
      allSelected={selectType === 'checkbox' && selectedItems.length === numItems}
      partialSelected={selectType === 'checkbox' && selectedItems.length > 0 && selectedItems.length < numItems}
    >
      <VirtualTableGrid.ColumnHeader
        id="title"
        sortable
        isSorted={sortField === 'title'}
        isAscending={isAscending}
        {...titleColSizes}
      >
        Title
      </VirtualTableGrid.ColumnHeader>
      <VirtualTableGrid.ColumnHeader
        id="description"
        isSorted={sortField === 'description'}
        isAscending={isAscending}
        {...descrColSizes}
      >
        Description
      </VirtualTableGrid.ColumnHeader>
      <VirtualTableGrid.ColumnHeader
        id="hosts"
        sortable
        isSorted={sortField === 'hosts'}
        isAscending={isAscending}
        {...countColSizes}
      >
        Hosts
      </VirtualTableGrid.ColumnHeader>
      <VirtualTableGrid.ColumnHeader
        id="clusters"
        sortable
        isSorted={sortField === 'clusters'}
        isAscending={isAscending}
        onSortToggle={() => this.onSortToggle('clusters')}
        {...countColSizes}
      >
        Clusters
      </VirtualTableGrid.ColumnHeader>
    </VirtualTableGrid.Head>
  );
};

const ItemRow = rowProps => {
  const { obj, tableData, index } = rowProps;
  const { selectType, selectedItem, selectedField, selectedItems } = tableData;
  const selected = selectType === 'checkbox' ? selectedItems.indexOf(obj) >= 0 : selectedItem === obj;
  return (
    <VirtualTableGrid.Row key={index} selected={(selectType === 'row' || selectType === 'checkbox') && selected}>
      <VirtualTableGrid.Col
        {...titleColSizes}
        selected={selectType === 'cell' && selected && selectedField === 'title'}
      >
        {obj.title}
      </VirtualTableGrid.Col>
      <VirtualTableGrid.Col
        {...descrColSizes}
        selected={selectType === 'cell' && selected && selectedField === 'description'}
      >
        {obj.description}
      </VirtualTableGrid.Col>
      <VirtualTableGrid.Col
        {...countColSizes}
        selected={selectType === 'cell' && selected && selectedField === 'hosts'}
      >
        {obj.hosts}
      </VirtualTableGrid.Col>
      <VirtualTableGrid.Col
        {...countColSizes}
        selected={selectType === 'cell' && selected && selectedField === 'clusters'}
      >
        {obj.clusters}
      </VirtualTableGrid.Col>
    </VirtualTableGrid.Row>
  );
};

test('VirtualTableGrid renders properly', () => {
  const tableData = {
    sortField: 'title',
    isAscending: true,
    numItems: mockVirtualItems.length
  };

  const component = mount(
    <VirtualTableGrid
      id="virtual-table-grid"
      header={ItemHeader}
      row={ItemRow}
      data={mockVirtualItems}
      tableData={tableData}
    />
  );
  expect(component.render()).toMatchSnapshot();
});

test('Bordered TableGrid renders properly', () => {
  const tableData = {
    sortField: 'title',
    isAscending: true,
    numItems: mockVirtualItems.length
  };

  const component = mount(
    <VirtualTableGrid
      id="virtual-table-grid"
      bordered
      header={ItemHeader}
      row={ItemRow}
      data={mockVirtualItems}
      tableData={tableData}
    />
  );
  expect(component.render()).toMatchSnapshot();
});

test('TableGrid with Row select renders properly', () => {
  const tableData = {
    sortField: 'title',
    isAscending: true,
    selectType: 'row',
    selectedItem: mockVirtualItems[0],
    numItems: mockVirtualItems.length
  };

  const component = mount(
    <VirtualTableGrid
      id="virtual-table-grid"
      bordered
      header={ItemHeader}
      row={ItemRow}
      data={mockVirtualItems}
      tableData={tableData}
    />
  );
  expect(component.render()).toMatchSnapshot();
});

test('TableGrid with Cell select renders properly', () => {
  const tableData = {
    sortField: 'title',
    isAscending: true,
    selectType: 'cell',
    selectedItem: mockVirtualItems[0],
    selectedField: 'hosts',
    numItems: mockVirtualItems.length
  };

  const component = mount(
    <VirtualTableGrid
      id="virtual-table-grid"
      bordered
      header={ItemHeader}
      row={ItemRow}
      data={mockVirtualItems}
      tableData={tableData}
    />
  );
  expect(component.render()).toMatchSnapshot();
});

test('TableGrid with Checkbox select and no selections renders properly', () => {
  const tableData = {
    sortField: 'title',
    isAscending: true,
    selectType: 'checkbox',
    selectedItems: [],
    numItems: mockVirtualItems.length
  };

  const component = mount(
    <VirtualTableGrid
      id="virtual-table-grid"
      bordered
      header={ItemHeader}
      row={ItemRow}
      data={mockVirtualItems}
      tableData={tableData}
    />
  );
  expect(component.render()).toMatchSnapshot();
});

test('TableGrid with Checkbox select and partial selections renders properly', () => {
  const tableData = {
    sortField: 'title',
    isAscending: true,
    selectType: 'checkbox',
    selectedItems: [mockVirtualItems[0]],
    numItems: mockVirtualItems.length
  };

  const component = mount(
    <VirtualTableGrid
      id="virtual-table-grid"
      bordered
      header={ItemHeader}
      row={ItemRow}
      data={mockVirtualItems}
      tableData={tableData}
    />
  );
  expect(component.render()).toMatchSnapshot();
});

test('TableGrid with Checkbox select and everything selected renders properly', () => {
  const tableData = {
    sortField: 'title',
    isAscending: true,
    selectType: 'checkbox',
    selectedItems: [...mockVirtualItems],
    numItems: mockVirtualItems.length
  };

  const component = mount(
    <VirtualTableGrid
      id="virtual-table-grid"
      bordered
      header={ItemHeader}
      row={ItemRow}
      data={mockVirtualItems}
      tableData={tableData}
    />
  );
  expect(component.render()).toMatchSnapshot();
});
