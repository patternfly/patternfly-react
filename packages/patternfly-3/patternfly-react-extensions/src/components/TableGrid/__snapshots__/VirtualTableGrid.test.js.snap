// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bordered TableGrid renders properly 1`] = `
<div
  class="table-grid-pf container-fluid bordered"
  id="virtual-table-grid"
>
  <div
    class="table-grid-pf-head row"
  >
    <div
      class="table-grid-pf-column-header text-nowrap active-sort col-md-2 col-sm-4 col-xs-6"
      id="title"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Title
        <span
          aria-hidden="true"
          class="pficon pficon-sort-common-asc table-grid-pf-header-sort-arrow"
        />
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-6 col-sm-4 col-xs-6"
      id="description"
    >
      Description
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="hosts"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Hosts
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="clusters"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Clusters
      </button>
    </div>
  </div>
  <div
    class="table-grid-pf-body"
  >
    <div
      style="overflow: visible; width: 0px;"
    >
      <div>
        <div
          aria-label="grid"
          aria-readonly="true"
          class="ReactVirtualized__Grid ReactVirtualized__List"
          role="grid"
          style="box-sizing: border-box; direction: ltr; height: auto; position: relative; width: 0px; overflow-x: hidden; overflow-y: auto;"
        />
      </div>
    </div>
    <div
      class="resize-triggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>
</div>
`;

exports[`TableGrid with Cell select renders properly 1`] = `
<div
  class="table-grid-pf container-fluid bordered"
  id="virtual-table-grid"
>
  <div
    class="table-grid-pf-head row"
  >
    <div
      class="table-grid-pf-column-header text-nowrap active-sort col-md-2 col-sm-4 col-xs-6"
      id="title"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Title
        <span
          aria-hidden="true"
          class="pficon pficon-sort-common-asc table-grid-pf-header-sort-arrow"
        />
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-6 col-sm-4 col-xs-6"
      id="description"
    >
      Description
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="hosts"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Hosts
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="clusters"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Clusters
      </button>
    </div>
  </div>
  <div
    class="table-grid-pf-body"
  >
    <div
      style="overflow: visible; width: 0px;"
    >
      <div>
        <div
          aria-label="grid"
          aria-readonly="true"
          class="ReactVirtualized__Grid ReactVirtualized__List"
          role="grid"
          style="box-sizing: border-box; direction: ltr; height: auto; position: relative; width: 0px; overflow-x: hidden; overflow-y: auto;"
        />
      </div>
    </div>
    <div
      class="resize-triggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>
</div>
`;

exports[`TableGrid with Checkbox select and everything selected renders properly 1`] = `
<div
  class="table-grid-pf container-fluid bordered"
  id="virtual-table-grid"
>
  <div
    class="table-grid-pf-head row"
  >
    <div
      class="table-grid-pf-checkbox"
    >
      <button
        aria-label="Click to deselect all"
        class="btn btn-link"
        type="button"
      >
        <span
          aria-hidden="true"
          class="fa fa-check-square"
        />
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap active-sort col-md-2 col-sm-4 col-xs-6"
      id="title"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Title
        <span
          aria-hidden="true"
          class="pficon pficon-sort-common-asc table-grid-pf-header-sort-arrow"
        />
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-6 col-sm-4 col-xs-6"
      id="description"
    >
      Description
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="hosts"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Hosts
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="clusters"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Clusters
      </button>
    </div>
  </div>
  <div
    class="table-grid-pf-body"
  >
    <div
      style="overflow: visible; width: 0px;"
    >
      <div>
        <div
          aria-label="grid"
          aria-readonly="true"
          class="ReactVirtualized__Grid ReactVirtualized__List"
          role="grid"
          style="box-sizing: border-box; direction: ltr; height: auto; position: relative; width: 0px; overflow-x: hidden; overflow-y: auto;"
        />
      </div>
    </div>
    <div
      class="resize-triggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>
</div>
`;

exports[`TableGrid with Checkbox select and no selections renders properly 1`] = `
<div
  class="table-grid-pf container-fluid bordered"
  id="virtual-table-grid"
>
  <div
    class="table-grid-pf-head row"
  >
    <div
      class="table-grid-pf-checkbox"
    >
      <button
        aria-label="Click to select all"
        class="btn btn-link"
        type="button"
      >
        <span
          aria-hidden="true"
          class="fa fa-square-o"
        />
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap active-sort col-md-2 col-sm-4 col-xs-6"
      id="title"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Title
        <span
          aria-hidden="true"
          class="pficon pficon-sort-common-asc table-grid-pf-header-sort-arrow"
        />
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-6 col-sm-4 col-xs-6"
      id="description"
    >
      Description
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="hosts"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Hosts
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="clusters"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Clusters
      </button>
    </div>
  </div>
  <div
    class="table-grid-pf-body"
  >
    <div
      style="overflow: visible; width: 0px;"
    >
      <div>
        <div
          aria-label="grid"
          aria-readonly="true"
          class="ReactVirtualized__Grid ReactVirtualized__List"
          role="grid"
          style="box-sizing: border-box; direction: ltr; height: auto; position: relative; width: 0px; overflow-x: hidden; overflow-y: auto;"
        />
      </div>
    </div>
    <div
      class="resize-triggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>
</div>
`;

exports[`TableGrid with Checkbox select and partial selections renders properly 1`] = `
<div
  class="table-grid-pf container-fluid bordered"
  id="virtual-table-grid"
>
  <div
    class="table-grid-pf-head row"
  >
    <div
      class="table-grid-pf-checkbox"
    >
      <button
        aria-label="Click to deselect all"
        class="btn btn-link"
        type="button"
      >
        <span
          aria-hidden="true"
          class="fa fa-minus-square"
        />
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap active-sort col-md-2 col-sm-4 col-xs-6"
      id="title"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Title
        <span
          aria-hidden="true"
          class="pficon pficon-sort-common-asc table-grid-pf-header-sort-arrow"
        />
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-6 col-sm-4 col-xs-6"
      id="description"
    >
      Description
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="hosts"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Hosts
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="clusters"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Clusters
      </button>
    </div>
  </div>
  <div
    class="table-grid-pf-body"
  >
    <div
      style="overflow: visible; width: 0px;"
    >
      <div>
        <div
          aria-label="grid"
          aria-readonly="true"
          class="ReactVirtualized__Grid ReactVirtualized__List"
          role="grid"
          style="box-sizing: border-box; direction: ltr; height: auto; position: relative; width: 0px; overflow-x: hidden; overflow-y: auto;"
        />
      </div>
    </div>
    <div
      class="resize-triggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>
</div>
`;

exports[`TableGrid with Row select renders properly 1`] = `
<div
  class="table-grid-pf container-fluid bordered"
  id="virtual-table-grid"
>
  <div
    class="table-grid-pf-head row"
  >
    <div
      class="table-grid-pf-column-header text-nowrap active-sort col-md-2 col-sm-4 col-xs-6"
      id="title"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Title
        <span
          aria-hidden="true"
          class="pficon pficon-sort-common-asc table-grid-pf-header-sort-arrow"
        />
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-6 col-sm-4 col-xs-6"
      id="description"
    >
      Description
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="hosts"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Hosts
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="clusters"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Clusters
      </button>
    </div>
  </div>
  <div
    class="table-grid-pf-body"
  >
    <div
      style="overflow: visible; width: 0px;"
    >
      <div>
        <div
          aria-label="grid"
          aria-readonly="true"
          class="ReactVirtualized__Grid ReactVirtualized__List"
          role="grid"
          style="box-sizing: border-box; direction: ltr; height: auto; position: relative; width: 0px; overflow-x: hidden; overflow-y: auto;"
        />
      </div>
    </div>
    <div
      class="resize-triggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>
</div>
`;

exports[`VirtualTableGrid renders properly 1`] = `
<div
  class="table-grid-pf container-fluid bordered"
  id="virtual-table-grid"
>
  <div
    class="table-grid-pf-head row"
  >
    <div
      class="table-grid-pf-column-header text-nowrap active-sort col-md-2 col-sm-4 col-xs-6"
      id="title"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Title
        <span
          aria-hidden="true"
          class="pficon pficon-sort-common-asc table-grid-pf-header-sort-arrow"
        />
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-6 col-sm-4 col-xs-6"
      id="description"
    >
      Description
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="hosts"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Hosts
      </button>
    </div>
    <div
      class="table-grid-pf-column-header text-nowrap col-md-2 col-sm-2 hidden-xs"
      id="clusters"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        Clusters
      </button>
    </div>
  </div>
  <div
    class="table-grid-pf-body"
  >
    <div
      style="overflow: visible; width: 0px;"
    >
      <div>
        <div
          aria-label="grid"
          aria-readonly="true"
          class="ReactVirtualized__Grid ReactVirtualized__List"
          role="grid"
          style="box-sizing: border-box; direction: ltr; height: auto; position: relative; width: 0px; overflow-x: hidden; overflow-y: auto;"
        />
      </div>
    </div>
    <div
      class="resize-triggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>
</div>
`;
